src/main/kotlin/org/jetbrains/tinygoplugin/TinyGoPluginConfiguration.kt

```diff
@@ -0,0 +1,82 @@
+package org.jetbrains.tinygoplugin
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.openapi.components.StoragePathMacros
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.util.xmlb.XmlSerializerUtil
+
+enum class GarbageCollector(val cmd: String) {
+    NONE("none"),
+    LEAKING("leaking"),
+    EXTALLOC("extalloc"),
+    CONSERVATIVE("conservative")
+}
+
+enum class Scheduler(val cmd: String) {
+    NONE("none"),
+    COROUTINES("coroutines"),
+    TASKS("tasks")
+}
+
+interface UserConfiguration {
+
+    var tinyGoSDKPath: String
+}
+
+interface ProjectConfiguration {
+    var gc: GarbageCollector
+    var scheduler: Scheduler
+    var targetPlatform: String
+    var goTags: String
+    var goArch: String
+    var goOS: String
+}
+
+@State(name = "TinyGoPluginUserConfig", storages = [Storage(StoragePathMacros.PRODUCT_WORKSPACE_FILE)])
+@Service(Service.Level.PROJECT)
+class UserConfigurationImpl : PersistentStateComponent<UserConfigurationImpl>, UserConfiguration {
+    override var tinyGoSDKPath = ""
+
+    override fun getState(): UserConfigurationImpl {
+        return this
+    }
+
+    override fun loadState(state: UserConfigurationImpl) {
+        XmlSerializerUtil.copyBean(state, this)
+    }
+}
+
+@State(name = "TinyGoPlugin", storages = [Storage(StoragePathMacros.WORKSPACE_FILE)])
+@Service(Service.Level.PROJECT)
+class ProjectConfigurationImpl :
+    PersistentStateComponent<ProjectConfigurationImpl>, ProjectConfiguration {
+    override var gc = GarbageCollector.NONE
+    override var scheduler = Scheduler.COROUTINES
```

Comment:
Conservative is probably a better default. In any case, GC and scheduler should be detected along with other settings. It doesn't happen now.